externals/llvm-external-projects/torch-mlir-dialects/lib/Conversion/TcpToLinalg/Elementwise.cpp

```diff
@@ -59,15 +59,56 @@ createLinalgPayloadForElementwiseOp(Operation *op,
                                     RankedTensorType resultTensorType,
                                     OpBuilder &b, ValueRange payloadArgs) {
   Location loc = op->getLoc();
+  auto elemType = resultTensorType.getElementType();
+
   if (isa<TanhOp>(op))
     return {b.create<math::TanhOp>(loc, payloadArgs[0])};
 
+  if (auto clampOp = dyn_cast<ClampOp>(op)) {
+    Value result = payloadArgs[0];
+    if (elemType.isa<mlir::FloatType>()) {
+      auto minFloat = clampOp.getMinFloat();
+      auto maxFloat = clampOp.getMaxFloat();
+      if (minFloat)
+        result = b.create<arith::MaxFOp>(
```

Comment:
I seem to recall that for degenerative floating point numbers the order of min and max matter (but can't recall details). What is expectation around denormals if any?

Reply:
Thats a good question. I haven't looked into those cases. The definition of `clamp` seems to be different in different frameworks. 
* PyTorch - max followed by min (https://pytorch.org/docs/stable/generated/torch.clamp.html)
* TensorFlow - min followed by max (https://github.com/tensorflow/tensorflow/blob/v2.11.0/tensorflow/python/ops/clip_ops.py#L113)

I just left a TODO for now to handle those cases.